src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs/breadcrumb.rs

```diff
@@ -0,0 +1,416 @@
+//! This module provides a clickable view for a single breadcrumb.
+
+use crate::prelude::*;
+
+use super::GLYPH_WIDTH;
+use super::HORIZONTAL_MARGIN;
+use super::VERTICAL_MARGIN;
+use super::TEXT_SIZE;
+
+use crate::graph_editor::MethodPointer;
```

Comment:
The math equations use `PADDING` in wrong way. For example, `separator_size = SEPARATOR_SIZE + PADDING`, and then `separator_size` is used as the width of thedrawing area. Padding should be added on both sides, not one! Please fix this, make screenshots, zoom in, and be sure that padding is applied correctly to all sides.

Reply:
This is also discussed above. Padding has a different purpose here. It isn't really used for position/alignment.

Reply:
@wdanilo `PADDING` was renamed to `SHAPE_PADDING`. It doesn't make any sense to have padding as we have margin and they wouldn't differ in functionality in our case.